### ISSUE TITLE
Ability to load the SDXL VAE FP16 Fix

### ISSUE THREAD

**Issue by boboij** [LostRuins/koboldcpp#845]
Add the ability to load the [SDXL VAE FP16 Fix from this repo](https://huggingface.co/madebyollin/sdxl-vae-fp16-fix) separately in koboldcpp (also from the GUI), as in stable-diffusion.cpp, to fix the problems with black images when running SDXL. The problem is described [in the stable-diffusion.cpp repo here](https://github.com/leejet/stable-diffusion.cpp?tab=readme-ov-file#features). The readme also shows how to load SDXL in stable-diffusion.cpp with the VAE Fix using a terminal [here](https://github.com/leejet/stable-diffusion.cpp?tab=readme-ov-file#txt2img-example).

**Comment by LostRuins** [comment]
Hi, can you link me to a model that doesn't work so I can test it?

**Comment by boboij** [comment]
How about [a link to the original SDXL model itself](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0.safetensors)? 
Here is the result after generating the image (black image instead of a cute cat):
![1234552](https://github.com/LostRuins/koboldcpp/assets/169610785/33e1b260-334e-46a7-a996-2ac226129237)
This can be fixed by loading the VAE Fix with an SDXL model. It fixes the problem with almost all SDXL models, I think. Stable-diffusion.cpp already has this capability.

**Comment by LostRuins** [comment]
Yes, I understand, that's because of the FP16 NAN like you mentioned. Do you have an example of a SD1.5 model with this issue? Reason being is that I want to see if I can swap VAEs automatically

**Comment by boboij** [comment]
I've never had this problem with SD1.5 models, so I don't have any examples.

**Comment by Noircogi** [comment]
Is there any SDXL model which currently works?  Every one I've tried results in black images.

**Comment by LostRuins** [comment]
Yes, the one in the wiki works.

**Comment by Amin456789** [comment]
taesdxl was fixing black images for me for sdxl turbo, not sure about the other models, plus it will decode imags much faster for cpus like mine.
but now these days i only use 1.5  lcm model so im out of the new changes for sdxl ones
